cmd/embedded-cluster/restore.go

```diff
@@ -0,0 +1,585 @@
+		return fmt.Errorf("unable to marshal config: %w", err)
+	}
+	fp, err := os.OpenFile(cfgpath, os.O_RDWR|os.O_CREATE, 0600)
+	if err != nil {
+		return fmt.Errorf("unable to create config file: %w", err)
+	}
+	defer fp.Close()
+	if _, err := fp.Write(data); err != nil {
+		return fmt.Errorf("unable to write config file: %w", err)
+	}
+	return nil
+}
+
+// runOutroForRestore calls Outro() in all enabled addons for restore operations by means of Applier.
+func runOutroForRestore(c *cli.Context) error {
+	os.Setenv("KUBECONFIG", defaults.PathToKubeConfig())
+	opts := []addons.Option{}
+	return addons.NewApplier(opts...).OutroForRestore(c.Context)
+}
+
+func isBackupRestorable(backup *velerov1.Backup, rel *release.ChannelRelease) (bool, string) {
+	if backup.Annotations["kots.io/embedded-cluster"] != "true" {
+		return false, "is not an embedded cluster backup"
+	}
+	if v := backup.Annotations["kots.io/embedded-cluster-version"]; v != defaults.Version {
+		return false, fmt.Sprintf("has a different embedded cluster version (%q) than the current version (%q)", v, defaults.Version)
+	}
+	if backup.Status.Phase != velerov1.BackupPhaseCompleted {
+		return false, fmt.Sprintf("has a status of %q", backup.Status.Phase)
+	}
+	if _, ok := backup.Annotations["kots.io/apps-versions"]; !ok {
+		return false, "is missing the kots.io/apps-versions annotation"
+	}
+	appsVersions := map[string]string{}
+	if err := json.Unmarshal([]byte(backup.Annotations["kots.io/apps-versions"]), &appsVersions); err != nil {
+		return false, "unable to json parse kots.io/apps-versions annotation"
+	}
+	if len(appsVersions) == 0 {
+		return false, "has no applications"
+	}
+	if len(appsVersions) > 1 {
+		return false, "has more than one application"
+	}
+	if _, ok := appsVersions[rel.AppSlug]; !ok {
+		return false, fmt.Sprintf("does not contain the %q application", rel.AppSlug)
+	}
+	if versionLabel := appsVersions[rel.AppSlug]; versionLabel != rel.VersionLabel {
+		return false, fmt.Sprintf("has a different app version (%q) than the current version (%q)", versionLabel, rel.VersionLabel)
+	}
+	return true, ""
+}
+
+// waitForBackups waits for backups to become available.
+// It returns a list of restorable backups, or an error if none are found.
+func waitForBackups(ctx context.Context) ([]velerov1.Backup, error) {
+	loading := spinner.Start()
+	defer loading.Close()
+	loading.Infof("Waiting for backups to become available")
+
+	cfg, err := k8sconfig.GetConfig()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get kubernetes config: %w", err)
+	}
+
+	veleroClient, err := veleroclientv1.NewForConfig(cfg)
+	if err != nil {
+		return nil, fmt.Errorf("unable to create velero client: %w", err)
+	}
+
+	rel, err := release.GetChannelRelease()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get release from binary: %w", err)
+	}
+	if rel == nil {
+		return nil, fmt.Errorf("no release found in binary")
+	}
+
+	for i := 0; i < 30; i++ {
+		time.Sleep(5 * time.Second)
```

Comment:
we should use expbackoff for this like we do in the util waits, or just an actual k8s wait

Reply:
Can update in a separate PR. This step usually only takes a few seconds anyways.